swarm/storage/resource.go

```diff
@@ -501,34 +507,36 @@ func (self *ResourceHandler) Update(name string, data []byte) (Key, error) {
 	}
 	version++
 
+	// create the update chunk
 	// prepend version and period to allow reverse lookups
 	// data header length does NOT include the header length prefix bytes themselves
 	headerlength := uint16(len(resource.nameHash) + 4 + 4)
-	fulldata := make([]byte, int(headerlength)+2+len(data))
+
+	key := self.resourceHash(resource.nameHash, nextperiod, version)
+	chunk := NewChunk(key, nil)
+	chunk.SData = make([]byte, signatureLength+int(headerlength)+2+len(data))
```

Comment:
It would be helpful for me to explain these calculations and numbers in them, like the `headerlength` calculation, or the length of SData here.